modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -1264,76 +1264,123 @@ func (suite *KeeperTestSuite) TestChanUpgradeCancel() {
-	suite.Require().True(ok)
+			err := path.EndpointB.ChanUpgradeTry()
+			suite.Require().NoError(err)
 
-	ctx := suite.chainA.GetContext()
-	suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.WriteUpgradeCancelChannel(ctx, path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, errorReceipt)
+			err = path.EndpointA.UpdateClient()
+			suite.Require().NoError(err)
 
-	channel = path.EndpointA.GetChannel()
+			errorReceipt, ok := suite.chainB.GetSimApp().IBCKeeper.ChannelKeeper.GetUpgradeErrorReceipt(suite.chainB.GetContext(), path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID)
+			suite.Require().True(ok)
 
-	// Verify that channel has been restored to previous state
-	suite.Require().Equal(types.OPEN, channel.State)
-	suite.Require().Equal(types.NOTINFLUSH, channel.FlushStatus)
-	suite.Require().Equal(mock.Version, channel.Version)
-	suite.Require().Equal(errorReceipt.Sequence, channel.UpgradeSequence)
+			ctx := suite.chainA.GetContext()
+			tc.malleate()
 
-	// Assert that state stored for upgrade has been deleted
-	upgrade, found := suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.GetUpgrade(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
-	suite.Require().Equal(types.Upgrade{}, upgrade)
-	suite.Require().False(found)
+			if tc.expPanic {
+				defer func() {
+					r := recover()
+					suite.NotNil(r, "The code did not panic")
+					suite.Equal(r, tc.messagePanic)
+				}()
+			}
+			suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.WriteUpgradeCancelChannel(ctx, path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, errorReceipt)
+			
+			channel = path.EndpointA.GetChannel()
 
-	// we need to find the event values from the proposed upgrade as the actual upgrade has been deleted.
-	proposedUpgrade := path.EndpointA.GetProposedUpgrade()
-	events := ctx.EventManager().Events().ToABCIEvents()
-	expEvents := ibctesting.EventsMap{
-		types.EventTypeChannelUpgradeCancel: {
-			types.AttributeKeyPortID:                path.EndpointA.ChannelConfig.PortID,
-			types.AttributeKeyChannelID:             path.EndpointA.ChannelID,
-			types.AttributeCounterpartyPortID:       path.EndpointB.ChannelConfig.PortID,
-			types.AttributeCounterpartyChannelID:    path.EndpointB.ChannelID,
-			types.AttributeKeyUpgradeConnectionHops: proposedUpgrade.Fields.ConnectionHops[0],
-			types.AttributeKeyUpgradeVersion:        proposedUpgrade.Fields.Version,
-			types.AttributeKeyUpgradeOrdering:       proposedUpgrade.Fields.Ordering.String(),
-			types.AttributeKeyUpgradeSequence:       fmt.Sprintf("%d", channel.UpgradeSequence),
-		},
-		sdk.EventTypeMessage: {
-			sdk.AttributeKeyModule: types.AttributeValueCategory,
-		},
-	}
+			// Verify that channel has been restored to previous state
+			suite.Require().Equal(types.OPEN, channel.State)
+			suite.Require().Equal(types.NOTINFLUSH, channel.FlushStatus)
```

Comment:
we'll be removing the FlushStatus field entirely so it's safe to remove this assertion from the test